source/extensions/filters/http/oauth2/oauth_client.h

```diff
@@ -38,8 +38,9 @@ class OAuth2Client : public Http::AsyncClient::Callbacks {
 
 class OAuth2ClientImpl : public OAuth2Client, Logger::Loggable<Logger::Id::upstream> {
 public:
-  OAuth2ClientImpl(Upstream::ClusterManager& cm, const envoy::config::core::v3::HttpUri& uri)
-      : cm_(cm), uri_(uri) {}
+  OAuth2ClientImpl(Upstream::ClusterManager& cm, const envoy::config::core::v3::HttpUri& uri,
+                   const absl::optional<std::chrono::seconds> default_expires_in)
```

Comment:
Is it okay to add `default_expires_in` to the client, or does it make more sense to let the client return an optional to the filter?

Reply:
I think this is fine